guidelines/frontend-coding-standard/wrap-imported-library-into-custom-component.md

```diff
@@ -0,0 +1,42 @@
+# Encourage to wrap imported libraries into custom components
+
+It is better to wrap the imported external libraries into the custom elements.
+
+It is always implemented in elements such as button and textfield. For example,
+
+```js
+import MaterialChip from "@mui/material/Chip";
+
+export const CHIP_TYPE = {
+  filled: "filled",
+  outlined: "outlined",
+};
+
+export function Chip({
+  label,
+  icon,
+  deleteIcon,
+  variant = CHIP_TYPE.outlined,
+  onIconClick,
+}) {
+  return (
+    <MaterialChip
+      deleteIcon={deleteIcon}
+      icon={icon}
+      label={label}
+      sx={{ borderRadius: 2, height: "32px" }}
+      variant={variant}
+      onDelete={onIconClick}
+    />
+  );
+}
+```
+
+The above is the example that wrapping the chip from Material UI into a custom chip component.
+
+---
+
+Implementing it has the following advantages:
+
+- It offers the flexibility of using the external elements in our own ways.
+- It helps us to easily change the codes if exteranl library is updated or depreciated, as there is only one position which imports the library.
```

Comment:
Add one more point like this:

- The public interface of 3rd party library usually is very large(since they have to support wilder and more general use cases). Wrapping up the library allow us to expose the only necessary interfaces in our codebase. This effectively reduce misuse and enforce a minimal governance on it. Just note we still have freedom to expose more public interface when necessary.